src/controller/java/src/chip/devicecontroller/model/InvokeElement.java

```diff
@@ -72,9 +111,30 @@ public String toString() {
         Locale.ENGLISH, "Endpoint %s, cluster %s, command %s", endpointId, clusterId, commandId);
   }
 
+  public static InvokeElement newInstance(
+      ChipPathId endpointId,
+      ChipPathId clusterId,
+      ChipPathId commandId,
+      byte[] tlv,
+      String jsonString) {
+    return new InvokeElement(
+        endpointId, clusterId, commandId, Optional.of(tlv), Optional.of(jsonString));
+  }
+
+  /** Create a new {@link InvokeElement} with only concrete ids. */
+  public static InvokeElement newInstance(
+      long endpointId, long clusterId, long commandId, byte[] tlv, String jsonString) {
```

Comment:
Similarly for the overloads provided for Optional vs non-optional, why not just annotate both parameters as `@Nullable`? They both end up resolving to similar behavior internally.

Reply:
updated with nullable, thanks

Reply:
Can you add @Nullable to the `tlv` and `jsonString` parameters for the two `newInstance` methods as well?

Reply:
done, thanks